src/lib/blockchain/wallet.ts

```diff
@@ -318,6 +322,66 @@ export class Wallet {
   public get web3Provider() {
     return this.web3 as Web3;
   }
+
+  public async getSFLForMatic(matic: string) {
```

Comment:
What is the purpose of the custom function encoding versus using the web3.contract to interact with this Uniswap router?

Reply:
> What is the purpose of the custom function encoding versus using the web3.contract to interact with this Uniswap router?

It's just there so we didn't need to store the whole ABI